### ISSUE TITLE
Error on date field with end-date

### ISSUE THREAD

**Issue by d0t15t** [olragon/graphql_api#13]
Hi,
we're really enjoying using graphql_api with drupal, thanks so much! 

Currently having an issue with date-fields which contain start and end dates - the schema is expecting a string, but receives an object (or array) and returns an error.

`"debugMessage": "Expected a value of type \"String\" but received: {\"value\":\"2018-07-06 00:00:00\",\"value2\":\"2018-07-30 00:00:00\",\"timezone\":\"Europe\\/Paris\",\"timezone_db\":\"Europe\\/Paris\",\"date_type\":\"datetime\"}",`

When using a date-field with a start date only, the query returns the datestamp as expected, but in this case it would be great to also access the timezone data.

Are you planning to support this? Or could you give any suggestion for how to add the correct schema data for all the date field info?

Thanks!

**Comment by olragon** [comment]
Post your GraphQL query and field config so I can dig deeper

**Comment by d0t15t** [comment]
```
{
  custom_entity__type_1 {
    title
    content_type_1 {
      nid
      title
      custom_entity__type_2 {
        title
        body
        date_field_with_end_date
      }
    }
  }
}

```
What do you mean exactly by field config? I've renamed the fields to better explain the entity and field types.

**Comment by d0t15t** [comment]
On a simpler query, the result is the same.
``` 
{
  node_article {
      field_test_date_field		
  }
}
```
`
"debugMessage": "Expected a value of type \"String\" but received: {\"value\":\"2016-12-31 23:00:00\",\"value2\":\"2018-05-31 22:00:00\",\"timezone\":\"Europe\\/Paris\",\"timezone_db\":\"UTC\",\"date_type\":\"datetime\"}",`

**Comment by olragon** [comment]
> What do you mean exactly by field config? I've renamed the fields to better explain the entity and field types.

You add a field date, you need select field type, field name, default value, timezone handling, start/end date .... What did you use for your field?

graphql_api resolve to string when we have no clue about schema types of given field type

**Comment by d0t15t** [comment]
> What did you use for your field?

I add a field of type 'date' (https://www.drupal.org/project/date). Add a field name, default value of 'now'. Timezone handling set to the same as the website. Set field to record year, month, day, hour, and minute. When using only start-date, graphql_api returns a datetime string. When set to also record an end-date, then graphql_api returns the error mentioned above.

Thank you for looking into it!

**Comment by Jehu** [comment]
@olragon any news here for using start/end date values in date field?

**Comment by olragon** [comment]
@Jehu I've been busy for the last week, try to fix in tomorrow

**Comment by olragon** [comment]
It should be fixed by c1590e894babeae9670bf99725f9c7f8f8f197fb

**Comment by Jehu** [comment]
@olragon many thanks. Date start and end are returned now. But there are new problems.
Because type 'struct' was deleted from Schema.php i've get this dump() output if i open /graphql
```
array:6 [▼
  "label" => "CER"
  "description" => "Information about the entity, used internally by CER."
  "type" => "struct"
  "getter callback" => "cer_get_cer_struct"
  "computed" => true
  "property info" => array:4 [▼
    "lineage" => array:5 [▼
      "label" => "Kontext"
      "description" => "The entity's lineage, represented as a string."
      "type" => "text"
      "getter callback" => "cer_get_entity_lineage"
      "computed" => true
    ]
    "depth" => array:5 [▼
      "label" => "Tiefe"
      "description" => "How deeply the entity is embedded."
      "type" => "integer"
      "getter callback" => "cer_get_entity_depth"
      "computed" => true
    ]
    "owner" => array:5 [▼
      "label" => "Eigentümer"
      "description" => "The top-level entity under which this one is embedded."
      "type" => "entity"
      "getter callback" => "cer_get_entity_owner"
      "computed" => true
    ]
    "original" => array:5 [▼
      "label" => "Original"
      "description" => "The original entity (before update), or the current entity if an update has not occurred."
      "type" => "entity"
      "getter callback" => "cer_get_entity_original"
      "computed" => true
    ]
  ]
]

Cannot detect fieldType for cer cer
```

If i add struct type again, i get a JSON formatted result with debug infos. I can use graphiQL also then. "cer" comes from https://www.drupal.org/project/cer which is used a lot in our project.

**Important:** There is a problem with text fields now:
```
query {
  fhp_shop_blocking {
    id
    field_blocking_reason {
      value
      format
    } 
    field_shop_blocking_dates {
      value
      value2
    }
  }
}
````
Result:
```
{
  "data": {
    "fhp_shop_blocking": [
      {
        "id": "1",
        "field_blocking_reason": {
          "value": null,
          "format": null
        },
        "field_shop_blocking_dates": {
          "value": "2018-08-01 00:00:00",
          "value2": "2018-08-25 00:00:00"
        }
      },
      {
        "id": "28",
        "field_blocking_reason": {
          "value": null,
          "format": null
        },
        "field_shop_blocking_dates": {
          "value": "2018-07-02 00:00:00",
          "value2": "2018-07-25 00:00:00"
        }
      }
    ]
  }
}
```

As you can see, the graphiQL autosuggests in query
```
[...]
    field_blocking_reason {
      value
      format
    } 
[...]
```
and they return now
```
[...]
        "field_blocking_reason": {
          "value": null,
          "format": null
        },
[...]
```

I appreciate your support. Please help us fix the remaining problems soon.
